core/engine.go

```diff
@@ -138,6 +138,15 @@ func (e *Engine) StartOutputs() error {
 			thresholdOut.SetThresholds(e.thresholds)
 		}
 
+		if thresholdOut, ok := out.(output.WithStopRunWithError); ok {
+			thresholdOut.SetStopRunWithErrorFunc(
```

Comment:
It has nothing to do with thresholds, right?
```suggestion
		if stopOut, ok := out.(output.WithStopRunWithError); ok {
			stopOut.SetStopRunWithErrorFunc(
```